### PR TITLE
Add Ark ASR 0.6B

### PR DESCRIPTION
**Pull request by wanglamao**
## Description

Adds Open ASR Leaderboard evaluation support for `AutoArk-AI/ARK-ASR-0.6B`.

This PR adds:

- `ark_asr/run_eval.py`: ARK-ASR evaluation entry point with batched inference.
- `ark_asr/run_ark_asr.sh`: local runner for the English short-form public benchmark splits.
- `ark_asr/submit_jobs_ark_asr.sh`: Hugging Face Jobs runner with configurable result bucket sync.
- `ark_asr/submit_jobs.sh`: compatibility wrapper for `submit_jobs_ark_asr.sh`.
- `requirements/requirements_ark_asr.txt`: additional dependencies for this runner.

The evaluation uses the 10 June 2026 setup from the `standardize_transformers` branch:

- Dataset: `hf-audio/open-asr-leaderboard`
- Runner: HF Jobs-compatible script with `FLAVOR=h200`
- Space: https://huggingface.co/spaces/AutoArk-AI/open-asr-leaderboard-ark-asr
- Public English short-form splits: `ami`, `earnings22`, `gigaspeech`, `librispeech/test.clean`, `librispeech/test.other`, `spgispeech`, `voxpopuli`
- Shared leaderboard utilities from `normalizer/data_utils.py` for data loading, normalization, and manifest writing
- Shared `normalizer.eval_utils.score_results(...)` for the final WER/RTFx summary

Results have been uploaded to the model repo:

https://huggingface.co/AutoArk-AI/ARK-ASR-0.6B/blob/main/.eval_results/open_asr_leaderboard.yaml

Results bucket/dataset for maintainer verification:

https://huggingface.co/datasets/AutoArk-AI/ark-asr-open-asr-leaderboard-results

## How to Run

Local smoke test:

```bash
pip install -r requirements/requirements.txt
pip install -r requirements/requirements_ark_asr.txt

cd ark_asr
MAX_EVAL_SAMPLES=64 bash run_ark_asr.sh
```

Full HF Jobs run:

```bash
RESULTS_BUCKET="<your-bucket>" HF_TOKEN=hf_... bash ark_asr/submit_jobs_ark_asr.sh
```

To bill jobs to an organization:

```bash
ORG_NAME="<org-name>" RESULTS_BUCKET="<your-bucket>" HF_TOKEN=hf_... bash ark_asr/submit_jobs_ark_asr.sh
```

## Type of Change

- [x] New model
- [ ] New dataset
- [ ] Bug fix
- [ ] New feature
- [ ] Other

## New Model Checklist

Results are reported on the HF Hub in `.eval_results/open_asr_leaderboard.yaml`.

https://huggingface.co/AutoArk-AI/ARK-ASR-0.6B/blob/main/.eval_results/open_asr_leaderboard.yaml

### HF Jobs Setup

- [x] Duplicated/adapted the official HF Jobs-compatible Transformers Space for ARK-ASR: https://huggingface.co/spaces/AutoArk-AI/open-asr-leaderboard-ark-asr
- [x] Added required dependencies for ARK-ASR evaluation.
- [x] Added `ark_asr/run_eval.py` with batched inference.
- [x] Added `ark_asr/submit_jobs_ark_asr.sh` pointing to the ARK-ASR Space and a configurable results bucket.

### Key Guidelines

- [x] Uses the same decoding hyperparameters across all datasets for `AutoArk-AI/ARK-ASR-0.6B`.
- [x] Provides an HF Jobs runner with default `FLAVOR=h200`.
- [x] Uses batched inference.
- [x] Uses `normalizer/data_utils.py` for data loading, normalization, and manifest writing.
- [x] Provides result manifests through a bucket/dataset for maintainer verification.

Batch size used for this run was `64` on each dataset. The run below was executed on 8x RTX 4090, and this batch size is expected to fit on H200.

- `ami`: 64
- `earnings22`: 64
- `gigaspeech`: 64
- `librispeech/test.clean`: 64
- `librispeech/test.other`: 64
- `spgispeech`: 64
- `voxpopuli`: 64

## Results

These results were generated on 8x RTX 4090, not on HF Jobs H200. The JSONL manifests are shared above so maintainers can re-score and/or reproduce them on H200.

Scored with:

```bash
PYTHONPATH=. python -c "from normalizer.eval_utils import score_results; score_results('ark_asr/results', 'AutoArk-AI/ARK-ASR-0.6B')"
```

Summary:

- Average WER: `5.97`
- Overall RTFx: `314.06`

| Split | WER | RTFx |
| --- | ---: | ---: |
| ami/test | 10.02 | 352.12 |
| earnings22/test | 9.77 | 331.88 |
| gigaspeech/test | 8.00 | 217.72 |
| librispeech/test.clean | 1.53 | 412.12 |
| librispeech/test.other | 3.51 | 382.56 |
| spgispeech/test | 2.63 | 352.83 |
| voxpopuli/test | 6.31 | 422.41 |

CSV summary:

```csv
model,RTFx,License,Size (B),# Languages,Encoder,Decoder,AMI WER,Earnings22 WER,Gigaspeech WER,LS Clean WER,LS Other WER,SPGISpeech WER,Voxpopuli WER
AutoArk-AI/ARK-ASR-0.6B,314.06,,,,,,10.02,9.77,8.0,1.53,3.51,2.63,6.31
```

## Verification

- `python -m py_compile ark_asr/run_eval.py normalizer/data_utils.py normalizer/eval_utils.py`
- `bash -n ark_asr/run_ark_asr.sh`
- `bash -n ark_asr/submit_jobs.sh`
- `bash -n ark_asr/submit_jobs_ark_asr.sh`
- `PYTHONPATH=. python -c "from normalizer.eval_utils import score_results; score_results('ark_asr/results', 'AutoArk-AI/ARK-ASR-0.6B')"`

## Related Issues

Closes #

### CODE REVIEW

File: ark_asr/run_eval.py [hunk 1]
```diff
@@ -0,0 +1,780 @@
...
+        )
+        write_manifest_records(
+            manifest_path,
+            all_results["references"],
+            all_results["predictions"],
+            all_results["audio_length_s"],
+            all_results["transcription_time_s"],
+        )
+    else:
+        manifest_path = data_utils.write_manifest(
```
Comment (ebezzam) [inline ark_asr/run_eval.py]: can you just rely on this common utility of writing the results?

File: ark_asr/run_eval.py [hunk 2]
```diff
@@ -0,0 +1,780 @@
...
+
+        with torch.no_grad():
+            outputs = self.model.generate(**inputs, **generate_kwargs)
+
+        input_ids = inputs["input_ids"]
+        predictions = []
+        for index, output in enumerate(outputs):
+            generated_ids = output[len(input_ids[index].tolist()) :]
+            prediction_raw = self.tokenizer.decode(generated_ids, skip_special_tokens=False)
+            predictions.append(normalize_prediction_text(prediction_raw))
```
Comment (ebezzam) [inline ark_asr/run_eval.py]: can you use the common normalizer?

**Review by ebezzam — COMMENTED** [review]
@wanglamao thank you for the contribution!

We're in the middle of switching to H200 + improving the normalizer + standardizing the eval process with #142 

To get your model addition ready for that transition, could you try this new setup?
- Setup: https://github.com/ebezzam/open_asr_leaderboard/tree/standardize_transformers#evaluate-a-model-as-of-10-june-2026
- New model checklist (model not in Transformers): https://github.com/ebezzam/open_asr_leaderboard/blob/standardize_transformers/.github/PULL_REQUEST_TEMPLATE.md#my-model-is-not-in-transformers-yet-

Could you also share a bucket/dataset with your results? as done here: https://github.com/huggingface/open_asr_leaderboard/pull/164#issuecomment-4698052192

Be sure to run the submit job script from this [branch/fork](https://github.com/ebezzam/open_asr_leaderboard/tree/standardize_transformers) so that you use the new normalizer!

Thank you! This will help us verify your model outputs 🤗

**Comment by wanglamao** [comment]
Thanks @ebezzam , I updated this PR for the `standardize_transformers` setup.

Changes in the latest commit:

- Added/updated the ARK-ASR HF Jobs runner: `ark_asr/submit_jobs_ark_asr.sh`
- Duplicated/adapted the official eval Space for ARK-ASR:
  https://huggingface.co/spaces/AutoArk-AI/open-asr-leaderboard-ark-asr
- Updated `ark_asr/run_eval.py` to use the shared `normalizer/data_utils.py` utilities for data loading, normalization, and manifest writing.
- Updated scoring to use the shared `normalizer.eval_utils.score_results(...)` path with `kaldialign` compound merging.
- Removed the old TED-LIUM result from the reported public short-form set.
- Uploaded raw JSONL manifests for all 7 public splits:
  https://huggingface.co/datasets/AutoArk-AI/ark-asr-open-asr-leaderboard-results
- Updated the model repo eval YAML:
  https://huggingface.co/AutoArk-AI/ARK-ASR-0.6B/blob/main/.eval_results/open_asr_leaderboard.yaml

One note for transparency: the shared manifests/results were generated on our own 8x RTX 4090 machine, not through HF Jobs/H200. The evaluation follows the updated leaderboard logic: 7 public splits, raw JSONL manifests, shared `normalizer/data_utils.py`, and `normalizer.eval_utils.score_results(...)` with the new English scoring path.

I also added the HF Jobs Space and `submit_jobs_ark_asr.sh` so the run can be reproduced through HF Jobs. Could you confirm whether an HF Jobs/H200 run is strictly required for this PR, or whether the provided raw manifests plus the standardized scorer are sufficient for you to re-score/verify?

File: ark_asr/run_eval.py (hunk 2 above)
Comment (wanglamao) [reply ark_asr/run_eval.py]: thank you for the instruction, it is now properly handled

File: ark_asr/run_eval.py (hunk 1 above)
Comment (wanglamao) [reply ark_asr/run_eval.py]: thank you for the instruction, it is now properly handled

**Comment by ebezzam** [comment]
@wanglamao thanks for the quick changes and preparing the HF space! It helped a lot to verify your results.

I had to make small changes (to use soundfile as a backend) and did some other simplifications in [this space](https://huggingface.co/spaces/hf-audio/open-asr-leaderboard-ark-asr/tree/main) if you want to check?

I obtained these results when scoring from the `standardize_transformers` branch/fork:

```
********************************************************************************
Results per dataset:
********************************************************************************
AutoArk/AI-ARK-ASR-0.6B | hf-audio-open-asr-leaderboard_ami_test: WER = 9.81 %, RTFx = 598.48
AutoArk/AI-ARK-ASR-0.6B | hf-audio-open-asr-leaderboard_earnings22_test: WER = 9.69 %, RTFx = 601.60
AutoArk/AI-ARK-ASR-0.6B | hf-audio-open-asr-leaderboard_gigaspeech_test: WER = 7.42 %, RTFx = 578.46
AutoArk/AI-ARK-ASR-0.6B | hf-audio-open-asr-leaderboard_librispeech_test.clean: WER = 1.47 %, RTFx = 635.01
AutoArk/AI-ARK-ASR-0.6B | hf-audio-open-asr-leaderboard_librispeech_test.other: WER = 3.44 %, RTFx = 585.18
AutoArk/AI-ARK-ASR-0.6B | hf-audio-open-asr-leaderboard_spgispeech_test: WER = 2.63 %, RTFx = 736.79
AutoArk/AI-ARK-ASR-0.6B | hf-audio-open-asr-leaderboard_voxpopuli_test: WER = 6.29 %, RTFx = 688.97

********************************************************************************
Composite Results:
********************************************************************************
AutoArk/AI-ARK-ASR-0.6B: WER = 5.82 %
AutoArk/AI-ARK-ASR-0.6B: RTFx = 672.17
********************************************************************************
```

Can you confirm the above results and metadata (below)?

model | License | Size (B) | # Languages | Encoder | Decoder
-- | -- | -- | -- | -- | --
AutoArk-AI/ARK-ASR-0.6B | apache-2.0 | 1.15 | 19 | Whisper | Qwen2

Note the number of parameters include that of the LLM decoder and is computed with [this](https://huggingface.co/spaces/hf-audio/open-asr-leaderboard-ark-asr/blob/main/run_eval.py#L477).

**Comment by wanglamao** [comment]
@ebezzam Thanks a lot for the updates!

I have checked the changes on the Hugging Face Space, and everything looks good to me. 
I confirm the benchmark results  and the metadata.

Thanks again for helping verify the model outputs!

**Comment by ebezzam** [comment]
@wanglamao thanks for confirming!

I'll leave this PR open, but I have the results ready and the submit jobs script in my branch for when we complete the transition.

**Comment by ebezzam** [comment]
@wanglamao closing this PR as #142 adds the model in the submit jobs script [here](https://github.com/huggingface/open_asr_leaderboard/blob/8d338ca0115a18f90d74e67b29efb480210bbd96/ark_asr/submit_jobs.sh#L14) and the result is live on the leaderboard!
